rewrite-java-test/src/test/java/org/openrewrite/java/cleanup/UseSystemLineSeparatorTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.Issue;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class UseSystemLineSeparatorTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new UseSystemLineSeparator());
+    }
+
+    @Issue("https://github.com/openrewrite/rewrite/issues/2363")
+    @Test
+    void useSystemLineSeparator() {
+        rewriteRun(
+          java(
+            """
+              class A {
+                  String s = System.getProperty("line.separator");
+                  var a = System.getProperty("line.separator");
+              }
+              """,
+            """
+              class A {
+                  String s = System.lineSeparator();
+                  var a = System.lineSeparator();
+              }
+              """
+          )
+        );
+    }
+    
+    
+    @Issue("https://github.com/openrewrite/rewrite/issues/2363")
+    @Test
+    void useSystemLineSeparatorStaticImport() {
+        rewriteRun(
+          java(
+            """
+              class A {
+                  import static java.lang.System.getProperty;
+                  String s = getProperty("line.separator");
+                  var a = getProperty("line.separator");
+              }
+              """,
+            """
+              class A {
+                  import static java.lang.System.getProperty;
```

Comment:
I think you meant to import `lineSeparator` here.

Reply:
I need some help here. I pushed 37e09e0b146101f72c532e54c9554ae36ceb5dbe but there is some issue because 

```java

final JavaTemplate template = JavaTemplate
                            .builder(this::getCursor, "lineSeparator()")
                            .staticImports("java.lang.System.lineSeparator")
                            .build();

```

Is doing the replacement but not adding the import as expected

Reply:
I also tried

```java

                    maybeAddImport("java.lang.System", "lineSeparator");


```

But it produces the same output

Reply:
So I think this is due to handling within the `AddInport` since `java.lang` is a default import in Java source code for non-static usages. However, it misses when a static import is used which is also a valid contruct.

https://github.com/openrewrite/rewrite/blob/7c361bc908222e1d47fb4caa1b3ebf2cbae0674f/rewrite-java/src/main/java/org/openrewrite/java/AddImport.java#L80

Reply:
It's true, this may be the very first time we've tried to statically import something from java.lang 😅